packages/mgt/src/graph/graph.presence.ts

```diff
@@ -55,30 +115,60 @@ export async function getUsersPresenceByPeople(graph: IGraph, people?: IDynamicP
   if (!people || people.length === 0) {
     return {};
   }
+
   const betaGraph = BetaGraph.fromGraph(graph);
   const batch = betaGraph.createBatch();
+  const peoplePresence = {};
+  let cache: CacheStore<CachePresence>;
+
+  if (presenceCacheEnabled()) {
+    cache = CacheService.getCache(cacheSchema, 'presence');
+  }
 
   for (const person of people) {
     if (person !== '' && person.id) {
       const id = person.id;
-      batch.get(id, `/users/${id}/presence`, ['presence.read', 'presence.read.all']);
+      peoplePresence[id] = null;
+      if (presenceCacheEnabled()) {
+        const presence = await cache.getValue(id);
+        if (presence && getPresenceInvalidationTime() > Date.now() - (await presence).timeCached) {
+          peoplePresence[id] = JSON.parse(presence.presence);
+        }
```

Comment:
You are not fetching the presence if not in the cache or if the cached presence is too old